hadoop-ozone/dist/dev-support/bin/dist-layout-stitching

```diff
@@ -114,6 +114,7 @@ run cp "${ROOT}/hadoop-ozone/objectstore-service/target/hadoop-ozone-objectstore
 cp -r "${ROOT}/hadoop-hdds/docs/target/classes/docs" ./
 
 #Copy docker compose files
-run cp -p -R "${ROOT}/hadoop-ozone/dist/src/main/compose" .
+#compose files are preprocessed: properties (eg. project.version) are replaced first by maven.
+run cp -p -R "${ROOT}/hadoop-ozone/dist/target/compose" .
```

Comment:
Can you elaborate on why we need to change cp compose files from src to target here? 
Do you mean to copy the filtered version of compose files after maven processing?

Reply:
Sure. It's a good question.

Until now we used wildcard in the docker-compose.yaml to define the classpath, independent from the current version:

```
HADOOP_CLASSPATH: /opt/ozone/share/ozone/lib/hadoop-ozone-filesystem-lib-current*.jar
```

Unfortunately it didn't work all the time. When I executed a command from the container (after `docker-compose exec scm`) the jar file is not added to the classpath.

I modified the docker-compose file to use the exact jar file. To make it version independent instead of the simple copy I replace the current version with maven (filtering) and copy to preprocessed version to the final destination with dist-layout-stitching script.

Old version: src/main/compose ---[copy with dist-layout-stitching]---> destination dir  
New version: src/main/compose ---[copy and replace version with maven]---> target/compose ---[dist layout stiching] --> destination dir

Reply:
Thanks for the details, that makes sense to me.